bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala

```diff
@@ -35,50 +35,54 @@ final case class ItcCcd(
     ampGain:                Double,     // the amplifier gain for this CCD (used to calculate ADU)
     warnings: List[ItcWarning]          // the warnings provided by ITC for this CCD
 ) {
-  val percentFullWell = peakPixelFlux / wellDepth * 100.0   // the max percentage of the well saturation for peak pixel
-  val adu             = (peakPixelFlux / ampGain).toInt     // the ADU value
+  val percentFullWell: Double = peakPixelFlux / wellDepth * 100.0   // the max percentage of the well saturation for peak pixel
+  val adu: Int                = (peakPixelFlux / ampGain).toInt     // the ADU value
 }
 
 sealed trait ItcResult extends Serializable {
-  def ccds:                         List[ItcCcd]
+  def ccds: List[ItcCcd]
 
-  def ccd(i: Int = 0)               = ccds(i % ccds.length)
-  def peakPixelFlux(ccdIx: Int = 0) = ccd(ccdIx).peakPixelFlux.toInt
+  /**
+    * Max value for a property across the CCDs
+    */
+  private def maxP[A](f: ItcCcd => A)(implicit a: scala.Ordering[A]): A = ccds.map(f).max
```

Comment:
Ah, right.  It'd have to be `f(ccds.maxBy(f))`.  Saves a pass though the `List`, I guess, but probably not worth it.